File: my-folder/0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> sumMap;
        vector<int> result;
        for (int i = 0; i < nums.size(); i++)
        {
            if (sumMap.find(nums.at(i)) != sumMap.end())
            {
                result.push_back(sumMap[nums.at(i)]);
                result.push_back(i);
                return result;
            }
            sumMap[target - nums.at(i)] = i;
        }
        return result;
    }
};

File: my-folder/0001-two-sum/solution.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        var map = HashMap<Int, Int>()
        var result = IntArray(2)
        for (i in nums.indices) {
            val complement = target - nums[i]
            if (map.containsKey(complement)) {
                result[0] = map[complement]!!
                result[1] = i
                return result
            }
            else
            {
                map.put(nums[i], i);
            }
        }
        return result
    }
}
